### PR TITLE
System indices auto-create for searchable-snapshots

### PR DESCRIPTION
**Pull request by pugnascotia**
Backport of #66276.

Part of #61656. Use the system indices auto-creation infrastructure
for the searchable snapshots plugin.

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java [hunk 1]
```diff
@@ -267,52 +121,37 @@ private static boolean isExpectedCacheGetException(Exception e) {
...
-                        }
-                    });
-                } catch (Exception e) {
-                    logger.warn(
-                        new ParameterizedMessage("cache fill failure: [{}]", CachedBlob.generateId(repository, name, path, offset)),
-                        e
-                    );
+                @Override
+                public void onFailure(Exception e) {
+                    logger.debug(new ParameterizedMessage("failure in cache fill: [{}]", request.id()), e);
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java:148]: We can use `debug(Supplier<?> msgSupplier, Throwable t)`

**Review by tlrx — APPROVED** [review]
LGTM but I've left minor comments

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java [hunk 2]
```diff
@@ -267,52 +121,37 @@ private static boolean isExpectedCacheGetException(Exception e) {
...
-
-            @Override
-            public void onFailure(Exception e) {
-                logger.error(() -> new ParameterizedMessage("failed to create blob cache system index [{}]", index), e);
-                listener.onFailure(e);
-            }
-        });
+            });
+        } catch (Exception e) {
+            logger.warn(new ParameterizedMessage("cache fill failure: [{}]", CachedBlob.generateId(repository, name, path, offset)), e);
```
Comment (tlrx) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/blobstore/cache/BlobStoreCacheService.java:153]: We should use warn(Supplier<?> msgSupplier, Throwable t)

**Comment by pugnascotia** [comment]
@tlrx thanks for the review. I won't change those logging methods as they are the same in `master`.

**Comment by tlrx** [comment]
Thanks. I think we can change them in master too.

**Pull request merged**